src/Trains.NET.Engine/Terrain/ITerrainMap.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+
+namespace Trains.NET.Engine.Tracks
+{
+    public interface ITerrainMap : IEnumerable<Terrain>
+    {
+        void AddTerrain(Terrain terrain);
+
+        Terrain GetAdjacentTerrainUp(Terrain terrain);
+        Terrain GetAdjacentTerrainDown(Terrain terrain);
+        Terrain GetAdjacentTerrainLeft(Terrain terrain);
+        Terrain GetAdjacentTerrainRight(Terrain terrain);
```

Comment:
You should have a look at whether TrackNeighbours can be made generic maybe

Reply:
As per discussion, leaving this for now and can iterate upon it.